File: pset6/readability.py
from cs50 import get_string

#prompt for string
s = get_string("Text: ").strip()
#define variables
letters, words, sentences = 0, 0, 0
#find all integers
for i in range(len(s)):
    if (i == 0 and s[i] != " ") or (i != len(s) - 1 and s[i] == " " and s[i +1 ] != " "):
        words +=1
    if s[i].isalpha():
        letters += 1
    if s[i] == "." or s[i] == "?" or s[i] == "!":
        sentences += 1
        
#define equation   
L = letters / words * 100
S = sentences / words * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

#print out grade level
if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <string.h> 
#include <math.h>

int main(void)
{
    //ask the reader for input
    string s = get_string("Text: ");
    //define the variables
    int letters, words, sentences; 
    letters = words = sentences = 0;
    //find all integers 
    for (int i = 0; i < strlen(s); i++)
    {
        char j = s[i];
        //find number of letters
        if (isalpha(j))
        {
            letters++;
        }
        //find number of words
        if (isspace(j))
        {
            words++;
        }
        //find number of sentences
        if (j == '.' || j == '?' || j == '!')
        {
            sentences++;
        }
    }
    words++;
    
    //define equation
    float L = (letters * 100) / words;
    float S = (sentences * 100) / words;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    
    //print out grade
    if (index < 1)
    
    {
        printf("Before Grade 1\n");
    }
    
    else if (index >= 16)
    
    {
        printf("Grade 16+\n");
    }
    
    else
    
    {
        printf("Grade %i\n", index);
    }
    
}
